### ISSUE TITLE
Repo's break after changing between private and public

### ISSUE THREAD

**Issue by PhilAtKitronik** [microsoft/pxt-microbit#4133]
We have had two repo that have had an issue between changing the repo from public to private (for development work) and then back to public.  The issue one repo was posted on the forum (link below) about blocks not being pulled into tutorials correctly.  When changes had been done, it would still not update and seem to pull in old versions of the code.

https://forum.makecode.com/t/tutorial-tool-blocks-not-appearing/8216?u=philatkitronik

After creating a new repo and copying the same code over into new files, the tutorials blocks all work fine.

The other repo that has had an issues, after doing the same switch between public > private > public, was unable to commit any work via the online integration.  This resulting having to create a new repo for this extenstion

The link to the two repo's are

https://github.com/KitronikLtd/pxt-kitronik-labbit

https://github.com/KitronikLtd/FAULTYpxt-kitronik-air-quality

@pelikhan @martinwork  @abchatra @AlasdairAtKitronik

**Comment by PhilAtKitronik** [comment]
any update with regards to this issue?

**Comment by abchatra** [comment]
We need tags to pull the latest one. You can see it is not tagged after 0.0.3 with changes.

https://github.com/KitronikLtd/FAULTYpxt-kitronik-air-quality/commits/master

**Comment by AlasdairAtKitronik** [comment]
> We need tags to pull the latest one. You can see it is not tagged after 0.0.3 with changes.
> 
> https://github.com/KitronikLtd/FAULTYpxt-kitronik-air-quality/commits/master

It's now been tagged and released at 0.0.4

**Comment by abchatra** [comment]
Is the issue fixed now?

**Comment by PhilAtKitronik** [comment]
Doing a release on https://github.com/KitronikLtd/pxt-kitronik-labbit has fixed that issue, although before we had done previous release and it was not working.  Has anything in the background changed that could effect this?

In terms of https://github.com/KitronikLtd/FAULTYpxt-kitronik-air-quality/commits/master it still seems to be broken and not able to import into Makecode.  Their is access to the repo, it just wont integrate into Makecode via the online editor or adding an extension via the github url

**Comment by pelikhan** [comment]
Private repos are not supported for extensions. This is by design.
________________________________

**Comment by PhilAtKitronik** [comment]
So if you cant switch or create private repo's until its ready for release and then switch it to the repo public, why is there an option when you create a github repo on the online editor?

![image](https://user-images.githubusercontent.com/37149430/123614576-615ac080-d7fc-11eb-96ee-767c6402efc4.png)

**Comment by pelikhan** [comment]
MakeCode will use your GitHub access to load a repository, so you are able to load and edit private repositories as long as you have access to them and you are signed in into the editor. 

However to share this repository as an extension, it has to be public as anyone is able to download an extension. We respect the access rights of your repository on GitHub.

**Comment by AlasdairAtKitronik** [comment]
@pelikhan 
But the issue we've faced is that the private repo becomes uneditable and unloadable after trying to switch from private to public, or even just after a random amount of time as a private repo.

We would want to do development of an extension as a private repo until we are ready to release a product to the world, otherwise people see what we're working on, and can start copying and changing our code (this happened with one of our current products). Surely it should be possible to do work in 'private mode' until you're ready to release and then switch to 'public mode', without it then causing breaking changes to repo access?

**Comment by pelikhan** [comment]
Indeed, there seems to be a regression in private repo support. @abchatra
